Tests/Clappr_Tests/Classes/Plugin/Playback/AVFoundationPlaybackQualityMetricsTests.swift

```diff
@@ -0,0 +1,117 @@
+                            kSourceUrl: "http://clappr.io/highline.mp4"
+                        ]
+                        let avfoundationPlayback = AVFoundationPlayback(options: options)
+                        let accessLog = AccessLogEventMock()
+                        accessLog.setIndicatedBitrate(7.0)
+                        let player = PlayerMock(accessLogEvent: accessLog)
+                        avfoundationPlayback.player = player
+                        avfoundationPlayback.addObservers()
+
+                        NotificationCenter.default.post(
+                            name: NSNotification.Name.AVPlayerItemNewAccessLogEntry,
+                            object: player.currentItem
+                        )
+
+                        expect(avfoundationPlayback.bitrate).toEventually(equal(7.0))
+                    }
+                }
+
+                context("with new observedBitrate value") {
+                    it("changes the bandwidth") {
+                        let options = [
+                            kSourceUrl: "http://clappr.io/highline.mp4"
+                        ]
+                        let avfoundationPlayback = AVFoundationPlayback(options: options)
+                        let accessLog = AccessLogEventMock()
+                        accessLog.setObservedBitrate(13.0)
+                        let player = PlayerMock(accessLogEvent: accessLog)
+                        avfoundationPlayback.player = player
+                        avfoundationPlayback.addObservers()
+
+                        NotificationCenter.default.post(
+                            name: NSNotification.Name.AVPlayerItemNewAccessLogEntry,
+                            object: player.currentItem
+                        )
+
+                        expect(avfoundationPlayback.bandwidth).toEventually(equal(13.0))
+                    }
+                }
+
+                context("with new numberOfDroppedVideoFrames value") {
+                    it("changes the droppedFrames") {
+                        let options = [
+                            kSourceUrl: "http://clappr.io/highline.mp4"
+                        ]
+                        let avfoundationPlayback = AVFoundationPlayback(options: options)
+                        let accessLog = AccessLogEventMock()
+                        accessLog.setDroppedFrames(2)
+                        let player = PlayerMock(accessLogEvent: accessLog)
+                        avfoundationPlayback.player = player
+                        avfoundationPlayback.addObservers()
+
+                        NotificationCenter.default.post(
+                            name: NSNotification.Name.AVPlayerItemNewAccessLogEntry,
+                            object: player.currentItem
+                        )
+
+                        expect(avfoundationPlayback.droppedFrames).toEventually(equal(2))
+                    }
+                }
+            }
+
+            context("when call play") {
```

Comment:
Make sense move te common code to the context and have only the "expect" inside each "it" ?

Reply:
I don't like this approach, in my opinion the test should have an explicit behavior. We follow the AAA pattern for tests and omitting code can brake this rule. What do you think?

Reply:
I think the context can be saw as part of the AAA pattern, I always imagine like I am reading a book when reading a test and the duplicity makes the read harder in this case to me.